modules/trusted-firmware-m/tfm_boards/common/attest_hal.c

```diff
@@ -132,7 +148,26 @@ enum tfm_plat_err_t tfm_plat_get_boot_seed(uint32_t size, uint8_t *buf)
 	if (nrf_err != NRF_CC3XX_PLATFORM_SUCCESS) {
 		return TFM_PLAT_ERR_SYSTEM_ERR;
 	}
+#elif defined(CONFIG_CRACEN_HW_PRESENT)
+	psa_status_t  psa_err = PSA_SUCCESS;
 
+	if (size != BOOT_SEED_SIZE) {
+		return TFM_PLAT_ERR_INVALID_INPUT;
+	}
+
+	if (!boot_seed_set) {
+		psa_err = psa_generate_random(boot_seed, sizeof(boot_seed));
+
+		if (psa_err != PSA_SUCCESS) {
+			return TFM_PLAT_ERR_SYSTEM_ERR;
+		}
+
+		boot_seed_set = true;
+	}
+	memcpy(buf, boot_seed, sizeof(uint8_t) * size);
```

Comment:
Why not just `sizeof(buf)`? It's better to use the size of `dest` instead of that of `src`. Or you can even do `MIN(sizeof(buf), size)`.

Reply:
Or well, I see that you check `size` above. At least you could remove the `sizeof(uint8_t)`, it's really just noise. And it could be nice to have the `size` check next to this `memcpy()`.